synapse/lib/agenda.py

```diff
@@ -442,21 +405,30 @@ def updateNexttime(self, now):
         if not self.recs:
             self._recidxnexttime = None
             self.nexttime = None
-            return
 
-    async def setDoc(self, text, nexs=False):
-        '''
-        Set the doc field of an appointment.
-        '''
-        self.doc = text
-        await self._save(nexs=nexs)
+        return self.nexttime
 
-    async def setName(self, text, nexs=False):
-        self.name = text
-        await self._save(nexs=nexs)
+    async def batch(self, edits):
+        for name, valu in edits.items():
+            if name not in self.__class__._synced_attrs:
+                extra = await self.stor.core.getLogExtra(name=name, valu=valu)
+                logger.warning('_Appt.batch() Invalid attribute received: %s = %r', name, valu, extra=extra)
+                continue
 
-    async def _save(self, nexs=False):
-        await self.stor._storeAppt(self, nexs=nexs)
+            if name == 'lasterrs':
+                # lasterrs is a deque and valu is a list
+                self.lasterrs.clear()
```

Comment:
If these will always be updated by clear/extend should we convert the lasterrs to a list and just let it fall through to the regular setattr?